src/stan/services/util/run_adaptive_sampler.hpp

```diff
@@ -87,6 +88,83 @@ void run_adaptive_sampler(Sampler& sampler, Model& model,
                           / 1000.0;
   writer.write_timing(warm_delta_t, sample_delta_t);
 }
+
+template <class Sampler, class Model, class RNG, typename SampT,
+          typename DiagnoseT>
+void run_adaptive_sampler(std::vector<Sampler>& samplers, Model& model,
+                          std::vector<std::vector<double>>& cont_vectors,
+                          int num_warmup, int num_samples, int num_thin,
+                          int refresh, bool save_warmup, std::vector<RNG>& rngs,
+                          callbacks::interrupt& interrupt,
+                          callbacks::logger& logger,
+                          std::vector<SampT>& sample_writers,
+                          std::vector<DiagnoseT>& diagnostic_writers,
+                          size_t n_chain) {
```

Comment:
Alright so the model, interrupt, and loggers passed into this all need to be threadsafe.

Is every function in the model threadsafe (I guess other than constructor, destructor)? And I would check this by inspecting a compiled model.hpp?

I see the test interrupt and logger have been made threadsafe.

It seems like as part of this pull there'd be a bunch of other interrupt/logger non-test code making things threadsafe. Is that not necessary for some reason? Or is it already threadsafe?